bin/upload.js

```diff
@@ -29,11 +29,18 @@ if (parts.length !== 2) {
     process.exit(1);
 }
 
+if (argv.name === true) {
+    console.log('error: please provide a name string when using the --name flag');
+    console.error(fs.readFileSync(__dirname + '/../USAGE.txt', 'utf8'));
+    process.exit(1);
+}
+
 var options = {
     account: account,
     mapid: dataset,
     accesstoken: process.env.MapboxAccessToken,
-    patch: argv.patch || undefined
+    patch: argv.patch || undefined,
+    name: argv.name || undefined
```

Comment:
Is there a good reason not to always provide the upload with a name? Maybe the `--name` flag would be nice to have if you want to override, but should we default to the filename if you didn't use the flag?